### PR TITLE
Consider objects from own namespace only in workload selector/less checkers.

### PR DESCRIPTION
**Pull request by hhovsepy**
Issue https://github.com/kiali/kiali/issues/5310
When checking "KIA0002 More than one selector-less object in the same namespace" and "KIA0003 More than one object applied to the same workload" for Sidecars, PeerAuthentications and RequestAuthentiations, it should show validation reference objects to only local namespace objects. Cross namespace validations are not supported for these object types.

Before: Sidecars, PeerAuths and RequestAuths validations of workload selectors "KIA0003" and "KIA0002" were considering objects from all namespaces.
Screenshot from 2022-07-19 14-43-24
Screenshot from 2022-07-19 14-40-49

Now: As those objects are not supporting cross-namespace validations, consider objects only from own namespace.
Screenshot from 2022-07-19 14-41-35
Screenshot from 2022-07-19 14-43-48

### CODE REVIEW

**Comment by jshaughn** [comment]
cc @israel-hdez looks like this one is still pending a review.

File: business/checkers/common/multi_match_selector_checker_test.go [hunk 1]
```diff
@@ -111,15 +127,15 @@ func TestSidecarsDifferentNamespaces(t *testing.T) {
...
 		}, data.CreateSidecar("sidecar2", "bookinfo2")),
 		data.AddSelectorToSidecar(map[string]string{
 			"app":     "details",
-			"version": "v1",
+			"version": "v3",
 		}, data.CreateSidecar("sidecar3", "bookinfo3")),
 		data.AddSelectorToSidecar(map[string]string{
 			"app":     "details",
-			"version": "v1",
+			"version": "v4",
```
Comment (israel-hdez) [inline business/checkers/common/multi_match_selector_checker_test.go:138]: Nitpick: if I understand correctly, `v4` does not exist in the list returned by `workloadList()`...
It is a very small nit. It does not affect the tests. So, take this as a simple comment.

So, no action needed.

If you do want to fix, perhaps do it on another PR as a side change (even if it is an unrelated change). Let's just merge this as is.

**Review by israel-hdez — APPROVED** [review]
Sorry for the delay.

Looks good!

File: business/checkers/common/multi_match_selector_checker_test.go (hunk 1 above)
Comment (hhovsepy) [reply business/checkers/common/multi_match_selector_checker_test.go:138]: actually I did it on purpose, to make sure that tests will fail if future changes cause bug

**Comment by hhovsepy** [comment]
thanks @israel-hdez
